OpenRA.Game/Map/MapCache.cs

```diff
@@ -39,6 +39,11 @@ public sealed class MapCache : IEnumerable<MapPreview>, IDisposable
 
 		public Dictionary<string, string> StringPool { get; } = new Dictionary<string, string>();
 
+		readonly List<MapDirectoryTracker> mapDirectoryTrackers = new List<MapDirectoryTracker>();
+		public string LastUpdatedUid;
+
+		public Action<string, string> MapUpdated = (x, y) => { };
```

Comment:
This should be an event handler: `public event Action<string, string> MapUpdated;`

Subscribers can do `MapUpdated += (...) => { ... };` to attach a handler to it.

The `MapCache` class is the only class that can invoke this by calling `MapUpdated(...)`. You won't be able to invoke this directly from the tracker. This is fine, because it means we're clearing up the ownership. I would recommend updating `LoadMap` to raise this event when it realises it is updating an existing entry. I would recommend adding a xmldoc comment to explain what the two parameters are for as well.

Reply:
I though it would be best to just keep it as action as I'm only implementing it in lobby. But I guess it wouldn't hurt to make it an event

Reply:
done

Reply:
I don't think `LoadMap` needs to raise this event, unless we want to raise it also when a map is added. Then we might as well call it when a map is removed